src/pages/PostDetails/PostDetails.jsx

```diff
@@ -1,26 +1,29 @@
 import { LikeOutlined } from '@ant-design/icons'
 import { IconText, Loading } from 'components'
-import { useEffect, useState } from 'react'
-import { useParams } from 'react-router-dom'
+import { useQuery } from 'react-query'
+import { useNavigate, useParams } from 'react-router-dom'
+import { postService } from 'services'
 
 const PostDetails = () => {
   const { postId } = useParams()
-  const [post, setPost] = useState()
-  const [loading, setLoading] = useState(true)
 
-  useEffect(() => {
-    fetch(`http://localhost:3000/posts/${postId}`)
-      .then((response) => response.json())
-      .then((data) => {
-        setPost(data)
-      })
-      .finally(() => setLoading(false))
-  }, [])
+  const {
+    isLoading,
+    data: post,
+    error,
+  } = useQuery(['post', postId], () => postService.fetchPost(postId))
+
+  const navigate = useNavigate()
+  const handleBackHome = () => navigate('/not-found')
```

Comment:
Function name suggests that you are navigated to hame, not to not found page, please change name

Reply:
And also not each error is related with not existing resource, so you need to handle navigation to not found page only when error is related with it